R/MltplxExperiment.R

```diff
@@ -179,24 +205,78 @@ as_MltplxExperiment.list = function(l, ...){
   if(any(n_classes > 1) ||
     length(unique_classes) > 1 ||
-    unique_classes[1] != "MltplxObject"){
+    !(unique_classes[1] %in% c("MltplxObject", "ppp"))){
     stop(paste('to convert "list" object to "MltplxObject" object, all',
-               'elements of list must be of class "MltplxObject"'))
+               'elements of list must either be of class "MltplxObject" or ',
+               'ppp'))
+  }
+
+  if(unique_classes[1] == "ppp"){
+    l = l %>%
+      furrr::future_imap(\(p, i){
+        new_MltplxObject(p$x, p$y, p$marks, paste("Slide", i),
+                         window = p$window)
+      })
+  }
+
+  # Check if list has smoothing/dist mats
+  pixel_size = NULL
+  unique_pixel_sizes = map_dbl(l, ~ ifelse(is.null(.x$mltplx_intensity$ps),
+                                           NA,
+                                           .x$mltplx_intensity$ps)) %>%
+    unique()
+  if(length(unique_pixel_sizes) > 1){
+    warning(paste0("`MltplxObjects` have different pixel sizes; ",
+                   "this information will be removed in the resulting ",
+                   "`MltplxExperiment` object"))
+  }else if(!is.na(unique_pixel_sizes[1])){
+    pixel_size = unique_pixel_sizes[1]
   }
 
-  x = map(l, ~ .x$mltplx_image$ppp$x) %>% unlist()
-  y = map(l, ~ .x$mltplx_image$ppp$y) %>% unlist()
-  marks = map(l, ~ .x$mltplx_image$ppp$marks) %>% unlist()
-  slide_ids = map(l, ~ rep(.x$slide_id, .x$mltplx_image$ppp$n)) %>% unlist()
-  window_sizes = tibble(
-    slide_id = map_chr(l, ~ .x$slide_id),
-    min_x = map_dbl(l, ~ .x$mltplx_image$ppp$window$xrange[1]),
-    max_x = map_dbl(l, ~ .x$mltplx_image$ppp$window$xrange[2]),
-    min_y = map_dbl(l, ~ .x$mltplx_image$ppp$window$yrange[1]),
-    max_y = map_dbl(l, ~ .x$mltplx_image$ppp$window$yrange[2])
+  bandwidth = NULL
+  unique_bws = map_dbl(l, ~ ifelse(is.null(.x$mltplx_intensity$bw),
+                                   NA,
+                                   .x$mltplx_intensity$bw)) %>%
+    unique()
+  if(length(unique_bws) > 1){
+    warning(paste0("`MltplxObjects` have different bandwidths; ",
+                   "this information will be removed in the resulting ",
+                   "`MltplxExperiment` object"))
+  }else if(!is.na(unique_bws[1])){
+    bandwidth = unique_bws[1]
+  }
+
+  dist_metric = NULL
+
+  unique_dist_metrics = map_chr(l, ~ ifelse(is.null(.x$mltplx_dist$metric),
+                                            NA,
+                                            .x$mltplx_dist$metric)) %>%
+    unique()
+  if(length(unique_dist_metrics) > 1){
+    warning(paste0("`MltplxObjects` have different distance metrics; ",
+                   "this information will be removed in the resulting ",
+                   "`MltplxExperiment` object"))
+  }else if(!is.na(unique_dist_metrics[1])){
+    dist_metric = unique_dist_metrics[1]
+  }
+
+  if(any(is.null(c(pixel_size, bandwidth, dist_metric)))){
+    pixel_size = NULL
+    bandwidth = NULL
+    dist_metric = NULL
```

Comment:
can we have non-null pixel size and bw, but null dist metric? Why do all of these have to be non-null?

Reply:
Yeah that's an error on my part- will fix